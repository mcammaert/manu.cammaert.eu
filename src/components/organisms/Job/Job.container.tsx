import React from 'react';

import { getJob } from 'api/resume';
import Job from './Job';

import { JobContainerProps } from './Job.types';

const JobContainer: React.FC<JobContainerProps> = ({ client: inputClient }) => {
  const job = getJob(inputClient);
  if (job) {
    const { client, description, startDate, endDate, roles, projects } = job;
    return <Job client={client} description={description} startDate={startDate} endDate={endDate} roles={roles} projects={projects} />;
  }
  return <></>;
};

export default JobContainer;
