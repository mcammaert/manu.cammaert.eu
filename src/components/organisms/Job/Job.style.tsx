import styled from 'styled-components';
import { modularScale, rem } from 'polished';

import { Copy } from 'components/atoms/Copy';

export const Job = styled.div`
  margin-bottom: ${modularScale(4)};
  position: relative;
  &:before {
    content: '';
    position: absolute;
    width: ${({ theme }) => rem(theme.margin)};
    height: 4px;
    top: -${modularScale(-2)};
    left: -${({ theme }) => rem(theme.margin)};
    background-color: ${({ theme }) => theme.colors.tertiary};
  }
`;

export const TitleContainer = styled.div`
  margin-bottom: ${modularScale(0)};
`;

export const Title = styled.div`
  font-size: ${modularScale(3)};
  line-height: ${modularScale(1)};
  font-weight: 200;
  color: ${({ theme }) => theme.colors.primary};
  margin: 0;
  margin-bottom: ${modularScale(-3)};
`;

export const Date = styled.div`
  font-size: ${modularScale(0)};
  font-weight: 400;
  color: ${({ theme }) => theme.colors.primary};
  margin-bottom: ${modularScale(2)};
  display: block;
`;

export const DateSegment = styled.span``;

export const Description = styled(Copy)`
  white-space: pre-line;
  margin-bottom: ${modularScale(2)};
`;

export const Label = styled.span``;

export const Roles = styled.ul`
  line-height: ${modularScale(1)};
  margin: 0;
  padding: 0;
  margin-bottom: ${modularScale(-1)};
  font-size: ${modularScale(1)};
  font-weight: 300;
`;

export const Role = styled.li`
  display: inline-block;

  &:after {
    content: '/';
    padding-left: ${modularScale(-4)};
    padding-right: ${modularScale(-4)};
  }
  &:last-child:after {
    content: '';
    padding 0;
  }
`;

export default { Job, TitleContainer, Title, Date, DateSegment, Description, Label, Roles, Role };
