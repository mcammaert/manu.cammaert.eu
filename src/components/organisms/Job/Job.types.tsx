import { Roles } from 'api/resume';

// eslint-disable-next-line @typescript-eslint/no-empty-interface
export interface JobProps {
  client: string;
  id: string;
  description?: string;
  startDate: string;
  endDate?: string;
  roles?: Roles;
  projects?: Array<{ name: string }>;
  displayProjects?: boolean;
}

export interface JobContainerProps {
  id: string;
}
