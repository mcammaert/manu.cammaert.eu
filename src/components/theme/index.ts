import { createGlobalStyle } from 'styled-components';
import 'normalize.css';
import { colors } from './colors';

export interface GlobalStyleProps {
  readonly light: boolean;
}

// const backgroundColor = (location: string) => (location === '/' ? colors.oxfordBlue : colors.fuzzyWuzzy);

export const GlobalStyle = createGlobalStyle<GlobalStyleProps>`
  body {
    -webkit-font-smoothing: antialiased;
    -moz-osx-font-smoothing: grayscale;
    font-family: 'Nunito Sans', sans-serif;
    width: 100vw;
    background: ${({ light }) => (light ? colors.white : colors.black)};
    background-attachment: fixed;    
    color: ${({ light }) => (light ? colors.black : colors.snow)};
    transition: all 800ms;
    a {
      color: ${({ theme }) => theme.colors.secondary};
    }
  }
`;

export const lightTheme = {
  margin: 15,
  colors: {
    primary: colors.charcoal,
    primaryInvert: colors.snow,
    secondary: colors.primary60,
    secondaryInvert: colors.white,
    tertiary: colors.secondary80,
    tertiaryInvert: colors.white,
    copy: colors.charcoal,
    copyInvert: colors.white,
  },
};

export const darkTheme = {
  margin: 15,
  colors: {
    primary: colors.snow,
    primaryInvert: colors.black,
    secondary: colors.primary30,
    secondaryInvert: colors.primary90,
    tertiary: colors.secondary30,
    tertiaryInvert: colors.secondary80,
    copy: colors.white,
    copyInvert: colors.charcoal,
  },
};
