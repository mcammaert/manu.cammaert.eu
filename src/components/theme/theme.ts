import { createGlobalStyle } from 'styled-components';

export const GlobalStyle = createGlobalStyle`
  body {
    background-color: #ccc;
  }
`;

const theme = {
  main: 'mediumseagreen',
};

export default theme;
