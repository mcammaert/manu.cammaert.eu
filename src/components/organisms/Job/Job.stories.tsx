import React from 'react';

import { text } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react';

import Job from './Job';

storiesOf('organisms/Job', module).add('Job', () => {
  const client = text('client', 'lorem ipsum');
  const id = text('id', 'lorem-ipsum');
  const description = text('description', 'lorem ipsum dolor sit amet');
  const startDate = text('start date', '2019-10-18');
  const endDate = text('end date', '2019-10-20');
  return <Job client={client} id={id} description={description} startDate={startDate} endDate={endDate} toggleProjectsHandler={() => {}} />;
});
