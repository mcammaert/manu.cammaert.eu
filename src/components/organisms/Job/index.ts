import * as S from './Job.style';

export { default as JobContainer } from './Job.container';
export { default as Job } from './Job';
export * from './Job.types';
export { S };
