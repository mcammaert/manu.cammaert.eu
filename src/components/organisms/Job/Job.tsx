import React from 'react';

import moment from 'moment';
import 'moment/locale/nl-be';

import { Container } from 'components/atoms/Container';
import { ScreenReaderOnly } from 'components/atoms/ScreenReaderOnly';
import { ProjectContainer } from 'components/organisms/Project';

import { JobProps } from './Job.types';

import S from './Job.style';

const Job: React.FC<JobProps> = ({ client, id, startDate, endDate, description, roles, projects, displayProjects = false }) => (
  <Container>
    <S.Job>
      <S.TitleContainer>
        <S.Title as="h3">{client}</S.Title>
      </S.TitleContainer>
      {roles && (
        <>
          <ScreenReaderOnly>
            <h4>Rollen</h4>
          </ScreenReaderOnly>
          <S.Roles>
            {roles.map(role => (
              <S.Role key={role}>{role}</S.Role>
            ))}
          </S.Roles>
        </>
      )}
      <S.Date as="span">
        {endDate && (
          <>
            <S.DateSegment>{`Van ${moment(startDate).format('LL')}`}</S.DateSegment>
            <S.DateSegment>{` tot ${moment(endDate).format('LL')}`}</S.DateSegment>
          </>
        )}
        {!endDate && (
          <>
            Sinds <S.DateSegment>{moment(startDate).format('LL')}</S.DateSegment>
          </>
        )}
      </S.Date>
      {description && (
        <>
          <ScreenReaderOnly>
            <h4>Omschrijving</h4>
          </ScreenReaderOnly>
          <S.Description>{description}</S.Description>
        </>
      )}
      {displayProjects && projects && (
        <>
          <ScreenReaderOnly>
            <h4>Projecten</h4>
          </ScreenReaderOnly>
          {projects.map(({ name }) => (
            <ProjectContainer key={name} client={id} project={name} />
          ))}
        </>
      )}
    </S.Job>
  </Container>
);

export default Job;
