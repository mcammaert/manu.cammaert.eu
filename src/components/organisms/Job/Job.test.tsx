import React from 'react';

import { cleanup } from '@testing-library/react';
import 'jest-styled-components';

import { renderComponent } from 'testing/react-testing-library';
import theme from 'components/theme/light';

import Job from './Job';

afterEach(cleanup);

describe('Job', () => {
  describe('default props', () => {
    it('should match the snapshot', () => {
      const component = <Job client="Lorem ipsum" description="Lorem ipsum dolor sit amet" startdate="2019-10-18" endDate="2019-10-20" />;
      const { container } = renderComponent({ theme, component });
      expect(container).toMatchSnapshot();
    });
  });
});
