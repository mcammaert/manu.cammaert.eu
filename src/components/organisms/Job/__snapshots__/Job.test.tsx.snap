// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Job default props should match the snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  width: 100vw;
  max-width: 800px;
  padding-left: 0.9375rem;
  padding-right: 0.9375rem;
}

.c6 {
  position: absolute;
  left: -10000px;
  top: auto;
  width: 1px;
  height: 1px;
  overflow: hidden;
}

.c7 {
  font-size: 1em;
  font-weight: 400;
  line-height: 1.776889em;
  color: #262830;
}

.c2 {
  margin-bottom: 3.157334518321em;
  position: relative;
}

.c2:before {
  content: '';
  position: absolute;
  width: 0.9375rem;
  height: 4px;
  top: -0.5627813555039173em;
  left: -0.9375rem;
  background-color: #d35239;
}

.c3 {
  margin-bottom: 1em;
}

.c4 {
  font-size: 2.3685930369999997em;
  line-height: 1.333em;
  font-weight: 200;
  color: #262830;
  margin: 0;
  margin-bottom: 0.42219156451906775em;
}

.c5 {
  font-size: 1em;
  font-weight: 400;
  color: #262830;
  margin-bottom: 1.776889em;
  display: block;
}

.c8 {
  white-space: pre-line;
  margin-bottom: 1.776889em;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <div
          class="c3"
        >
          <h3
            class="c4"
          >
            Lorem ipsum
          </h3>
        </div>
        <span
          class="c5"
        >
          <span
            class=""
          >
            Van 18 oktober 2019
          </span>
          <span
            class=""
          >
             tot 20 oktober 2019
          </span>
        </span>
        <div
          class="c6"
        >
          <h4>
            Omschrijving
          </h4>
        </div>
        <div
          class="c7 c8"
        >
          Lorem ipsum dolor sit amet
        </div>
      </div>
    </div>
  </div>
</div>
`;
